mm0-rs/src/parser.rs

```diff
@@ -223,13 +224,40 @@ impl<'a> Parser<'a> {
     r.unwrap_or_else(|e| self.errors.push(e))
   }
 
+  /// Parse and return a doc comment if one is present. If not, just return None.
+  fn doc_comment(&mut self) -> Option<DocComment> {
+    let mut doc = Vec::new();
+    while self.source.get(self.idx..).map(|slice_| slice_.starts_with(b"--|")).unwrap_or(false) {
+      // move past `--|`
+      self.idx += 3;
+      // take everything up to the end of the line.
+      while !(self.cur_opt() == Some(b'\n')) {
+        doc.push(self.cur());
```

Comment:
this will OOB if `self.cur_opt()` returns `None`.
```suggestion
      while self.cur_opt().map_or(false, |b| b == b'\n') {
        doc.push(self.cur());
```

Reply:
That's embarrassing. Nice catch.